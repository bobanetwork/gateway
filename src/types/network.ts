export type NetworkSwitchState =
  | 'idle'
  | 'confirming'
  | 'switching'
  | 'success'
  | 'failed'
  | 'chain-not-found'

export interface NetworkSwitchError {
  code?: number;
  message: string;
}
