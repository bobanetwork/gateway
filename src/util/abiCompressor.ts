/**
 * Script to prepare the human readable abi from the json.
 * based on the functions used out of the abi in network service.
 *
 * to prepare abi execute like
 * ```sh
 *  node ./util/abiCompressor.ts
 * ```
 *
 *
 */

import { ethers } from 'ethers'
import * as fs from 'fs'

const smart_contract_abi_paths = [
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/messaging/L2StandardBridge.sol/L2StandardBridge.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/standards/L2StandardERC20.sol/L2StandardERC20.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/predeploys/OVM_GasPriceOracle.sol/OVM_GasPriceOracle.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DiscretionaryExitFee.sol/DiscretionaryExitFee.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L1LiquidityPool.sol/L1LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/Teleportation.sol/Teleportation.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L2LiquidityPool.sol/L2LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/BobaFixedSavings.sol/BobaFixedSavings.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance-token/BOBA.sol/BOBA.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegate.sol/GovernorBravoDelegate.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegator.sol/GovernorBravoDelegator.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/L2BillingContract.sol/L2BillingContract.json',
  './src/deployment/contracts/L1ERC20.json',
  './src/deployment/contracts/NFTMonsterV2.json',
  './src/deployment/contracts/AuthenticatedFaucet.json',
  './src/deployment/contracts/WAGMIv0.json',
  './src/deployment/contracts/WAGMIv1.json',
  './src/deployment/contracts/crosschain/EthBridge.json',
  './src/deployment/contracts/crosschain/L2StandardERC20.json',
  './src/deployment/contracts/crosschain/LZEndpointMock.json',
]

const networkServicesStr = fs.readFileSync(
  './src/services/networkService.ts',
  'utf8'
)

const compress = () => {
  for (const path of smart_contract_abi_paths) {
    const rawData = fs.readFileSync(path, 'utf8')
    const abiObject = JSON.parse(rawData)
    const name = abiObject.contractName
    let abi = abiObject.abi
    abi = abi.filter((component) => {
      if (component['type'] === 'function' && component['name']) {
        if (networkServicesStr.includes(component['name'])) {
          return true
        }
        return false
      }
    })
    const abiInterface = new ethers.utils.Interface(abi)
    const humanReadableABI = abiInterface.format(ethers.utils.FormatTypes.full)
    const stringifiedHumanReadableABI = JSON.stringify(humanReadableABI)
    const result = `const ${name}ABI = ${stringifiedHumanReadableABI}\n export default ${name}ABI`
    fs.writeFile(`./src/services/abi/${name}.abi.ts`, result, () => {
      console.log(`wrote ${name}`)
    })
  }
}

;(() => {
  compress()
})()
