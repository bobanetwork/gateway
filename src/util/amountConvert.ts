/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js'
import { BigNumberish } from 'ethers'

export const logAmount = (
  amount: BigNumberish,
  power: number,
  truncate = 0
) => {
  // toString as some methods seem to supply BigNumber
  const x = new BigNumber(amount?.toString())
  const exp = new BigNumber(10).pow(power)

  const calculated = x.div(exp)

  if (truncate > 0) {
    return calculated.toFixed(truncate)
  } else {
    return calculated.toFixed()
  }
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export const powAmount = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export const toWei_String = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

const lookupMap = {
  ETH: { provider: 'ethereum' },
  BOBA: { provider: 'boba-network' },
  OLO: { provider: 'oolongswap' },
  OMG: { provider: 'omisego' },
  USDC: { provider: 'usd-coin' },
  BNB: { provider: 'binancecoin' },
  tBNB: { provider: 'binancecoin' },
}

export const amountToUsd = (amount: number, lookupPrice, token) => {
  const { symbol } = token
  if (!symbol) {
    return 0
  }
  const currencyInfo = lookupMap[symbol]
  if (!!currencyInfo && !!lookupPrice[currencyInfo.provider]) {
    return amount * lookupPrice[currencyInfo.provider].usd
  } else if (!!lookupPrice[symbol.toLowerCase()]) {
    return amount * lookupPrice[symbol.toLowerCase()].usd
  } else {
    return 0
  }
}

export const formatLargeNumber = (num: number) => {
  const exp = Math.floor(Math.log10(Math.abs(num)) / 3)
  const scaledNum = Math.abs(num) / Math.pow(10, exp * 3)
  const formattedNumber = scaledNum.toFixed(2)
  const exponent = ['', 'k', 'M', 'B', 'T'][exp] || ''
  return isNaN(Number(formattedNumber)) ? '0' : formattedNumber + exponent
}
