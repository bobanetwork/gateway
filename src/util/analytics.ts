import ReactGA from 'react-ga4'
import { GA4_MEASUREMENT_ID } from './constant'

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID?.toString())
  }
}

export const sendPageView = (path: string) => {
  ReactGA.send({
    hitType: 'pageview',
    page: path,
    title: path,
  })
}

export const trackClick = (category: string, action: string, label: string) => {
  ReactGA.event({
    category,
    action,
    label,
  })
}

export default {
  init,
  sendPageView,
  trackClick,
}
