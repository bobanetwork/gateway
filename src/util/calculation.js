/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair. 
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

var accAdd = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = (accMul(num1, times) + accMul(num2, times)) / times;
    return getCorrectResult("add", num1, num2, result);
};

var accSub = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var dec1, dec2, times;
    try { dec1 = countDecimals(num1)+1; } catch (e) { dec1 = 0; }
    try { dec2 = countDecimals(num2)+1; } catch (e) { dec2 = 0; }
    times = Math.pow(10, Math.max(dec1, dec2));
    var result = Number((accMul(num1, times) - accMul(num2, times)) / times);
    return getCorrectResult("sub", num1, num2, result);
};

var accDiv = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var t1 = 0, t2 = 0, dec1, dec2;
    try { t1 = countDecimals(num1); } catch (e) { }
    try { t2 = countDecimals(num2); } catch (e) { }
    dec1 = convertToInt(num1);
    dec2 = convertToInt(num2);
    var result = accMul((dec1 / dec2), Math.pow(10, t2 - t1));
    return getCorrectResult("div", num1, num2, result);
};

var accMul = function(num1, num2) {
    num1 = Number(num1);
    num2 = Number(num2);
    var times = 0, s1 = num1.toString(), s2 = num2.toString();
    try { times += countDecimals(s1); } catch (e) { }
    try { times += countDecimals(s2); } catch (e) { }
    var result = convertToInt(s1) * convertToInt(s2) / Math.pow(10, times);
    return getCorrectResult("mul", num1, num2, result);
};

var countDecimals = function(num) {
    var len = 0;
    try {
        num = Number(num);
        var str = num.toString().toUpperCase();
        if (str.split('E').length === 2) { // scientific notation
            var isDecimal = false;
            if (str.split('.').length === 2) {
                str = str.split('.')[1];
                if (parseInt(str.split('E')[0]) !== 0) {
                    isDecimal = true;
                }
            }
            let x = str.split('E');
            if (isDecimal) {
                len = x[0].length;
            }
            len -= parseInt(x[1]);
        } else if (str.split('.').length === 2) { // decimal
            if (parseInt(str.split('.')[1]) !== 0) {
                len = str.split('.')[1].length;
            }
        }
    } catch(e) {
        throw e;
    } finally {
        if (isNaN(len) || len < 0) {
            len = 0;
        }
        return len;
    }
};

var convertToInt = function(num) {
    num = Number(num);
    var newNum = num;
    var times = countDecimals(num);
    var temp_num = num.toString().toUpperCase();
    if (temp_num.split('E').length === 2) {
        newNum = Math.round(num * Math.pow(10, times));
    } else {
        newNum = Number(temp_num.replace(".", ""));
    }
    return newNum;
};

var getCorrectResult = function(type, num1, num2, result) {
    var temp_result = 0;
    switch (type) {
        case "add":
            temp_result = num1 + num2;
            break;
        case "sub":
            temp_result = num1 - num2;
            break;
        case "div":
            temp_result = num1 / num2;
            break;
        case "mul":
            temp_result = num1 * num2;
            break;
        default:
            break;
    }
    if (Math.abs(result - temp_result) > 1) {
        return temp_result;
    }
    return result;
};


export {
  accAdd,
  accSub,
  accDiv,
  accMul,
  countDecimals,
  convertToInt,
  getCorrectResult,
}