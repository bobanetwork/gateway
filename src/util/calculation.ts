/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair. 
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

const countDecimals = (num) => {
  let len = 0
  try {
    num = Number(num)
    let str = num.toString().toUpperCase()
    if (str.split('E').length === 2) {
      // scientific notation
      let isDecimal = false
      if (str.split('.').length === 2) {
        str = str.split('.')[1]
        if (parseInt(str.split('E')[0], 10) !== 0) {
          isDecimal = true
        }
      }
      const x = str.split('E')
      if (isDecimal) {
        len = x[0].length
      }
      len -= parseInt(x[1], 10)
    } else if (str.split('.').length === 2) {
      // decimal
      if (parseInt(str.split('.')[1], 10) !== 0) {
        len = str.split('.')[1].length
      }
    }
  } catch (e) {
    throw e
  } finally {
    if (isNaN(len) || len < 0) {
      len = 0
    }
  }
  return len
}

export { countDecimals }
