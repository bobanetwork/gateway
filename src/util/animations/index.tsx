import React, { useState, useEffect } from 'react'
import { TConductorInstance } from 'react-canvas-confetti/dist/types'
import Realistic from 'react-canvas-confetti/dist/presets/realistic'
import { TCanvasConfettiAnimationOptions } from 'react-canvas-confetti/dist/types/normalization'
import styled from 'styled-components'
import dayjs from 'dayjs'

const CanvaContainer = styled.div`
  display: block;
  z-index: 9999;
  position: fixed;
`
const count = 200
const defaults = {
  origin: { y: 0.7 },
}

export const Confetti = ({ size, particles, spreads }: any) => {
  const [conductor, setConductor] = useState<TConductorInstance>()

  const onInit = ({ conductor }: { conductor: TConductorInstance }) => {
    setConductor(conductor)
  }

  const onOnce = () => {
    conductor?.shoot()
  }

  const PARTICLES = {
    heart: [
      {
        type: 'path',
        path: 'M167 72c19,-38 37,-56 75,-56 42,0 76,33 76,75 0,76 -76,151 -151,227 -76,-76 -151,-151 -151,-227 0,-42 33,-75 75,-75 38,0 57,18 76,56z',
        matrix: [
          0.03333333333333333, 0, 0, 0.03333333333333333, -5.566666666666666,
          -5.533333333333333,
        ],
      },
    ],
    circle: ['circle'],
    star: ['star'],
  }

  const heartsOptions = (options: TCanvasConfettiAnimationOptions) => {
    return {
      ...options,
      startVelocity: 50,
      decay: 0.95,
      spread: 180,
      zIndex: 9999,
      gravity: 3,
      origin: {
        y: 1,
      },
      scalar: size || 7,
      particleCount: Math.floor(count * (Math.random() * (0.5 - 0.1) + 0.1)),
      colors: ['#AEDB01'],
      shapes: PARTICLES[particles] || ['square'],
    }
  }

  useEffect(() => {
    setTimeout(() => {
      onOnce()
    }, 1500)
  })

  return (
    <CanvaContainer>
      <Realistic onInit={onInit} decorateOptions={heartsOptions} />
    </CanvaContainer>
  )
}

export const ValentinesDayBanner = () => {
  const [isValentinesDay, setIsValentinesDay] = useState(false)

  useEffect(() => {
    const today = dayjs()
    const isFebruary14 = today.date() === 14 && today.month() === 1
    if (isFebruary14) {
      setIsValentinesDay(true)
    }
  }, [])
  return (
    <>
      {isValentinesDay && (
        <>
          <Confetti size={8} particles="heart" spreads={150} />
          <Confetti size={2} particles="heart" spreads={150} />
          <Confetti size={1} particles="heart" spreads={150} />
        </>
      )}
    </>
  )
}
