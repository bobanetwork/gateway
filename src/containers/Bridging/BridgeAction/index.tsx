import { setConnect } from 'actions/setupAction'
import { openModal } from 'actions/uiAction'
import { Heading } from 'components/global'
import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import {
  selectAccountEnabled,
  selectActiveNetworkType,
  selectAmountToBridge,
  selectBridgeAlerts,
  selectTokenToBridge,
} from 'selectors'
import { BridgeActionButton, BridgeActionContainer } from '../styles'
import { ONRAMP_API_KEY, PURCHASE_RAMP_URL } from 'util/constant'
import { NetworkType } from 'util/network/network.util'
import { trackClick } from 'util/analytics'

const BridgeAction = () => {
  const dispatch = useDispatch<any>()
  const accountEnabled = useSelector<any>(selectAccountEnabled())
  const token = useSelector(selectTokenToBridge())
  const amountToBridge = useSelector(selectAmountToBridge())
  const bridgeAlerts = useSelector(selectBridgeAlerts())

  const isMainnet =
    useSelector(selectActiveNetworkType()) === NetworkType.MAINNET

  const isBridgeActionDisabled = () => {
    const hasError = bridgeAlerts.find((alert: any) => alert.type === 'error')
    return !token || !amountToBridge || hasError
  }

  const onConnect = () => {
    dispatch(setConnect(true))
  }

  const onBridge = () => {
    if (!isBridgeActionDisabled()) {
      dispatch(openModal({ modal: 'bridgeConfirmModal' }))
    }
  }

  return (
    <BridgeActionContainer>
      {!accountEnabled ? (
        <BridgeActionButton
          onClick={onConnect}
          data-testid="connect-btn"
          label={<Heading variant="h3">Connect Wallet</Heading>}
        />
      ) : (
        <div
          style={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            width: '100%',
            gap: '10px',
          }}
        >
          <BridgeActionButton
            disabled={isBridgeActionDisabled()}
            onClick={onBridge}
            data-testid="bridge-btn"
            label={<Heading variant="h3">Bridge</Heading>}
          />
          {isMainnet && ONRAMP_API_KEY ? (
            <BridgeActionButton
              outline
              onClick={() => {
                trackClick('button', 'button_click', 'Buy With Ramp Network')
                window.open(PURCHASE_RAMP_URL, 'blank')
              }}
              data-testid="connect-btn"
              label={<Heading variant="h3">Buy with Ramp Network</Heading>}
            />
          ) : null}
        </div>
      )}
    </BridgeActionContainer>
  )
}

export default BridgeAction
