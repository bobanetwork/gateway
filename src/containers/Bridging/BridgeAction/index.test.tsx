import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { NETWORK, NETWORK_TYPE } from 'util/network/network.util'
import CustomThemeProvider from 'themes'
import BridgeAction from '.'
import thunk from 'redux-thunk'

const mockStore = configureStore([thunk])

const renderBridgeAction = ({ store }: any) => {
  return render(
    <Provider store={store}>
      <CustomThemeProvider>
        <BridgeAction />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('BridgeAction', () => {
  let store

  describe('AccountEnabled is true', () => {
    let alerts = [
      { text: 'alert1', type: 'info' },
      { text: 'alert2', type: 'info' },
    ]
    beforeEach(() => {
      store = mockStore({
        ui: {
          theme: 'dark',
        },
        network: {
          activeNetwork: NETWORK.ETHEREUM,
          activeNetworkType: NETWORK_TYPE.MAINNET,
        },
        bridge: {
          tokens: [{ symbol: 'ETH' }],
          amountToBridge: '1.255',
          alerts,
        },
        setup: {
          accountEnabled: true,
        },
      })
    })

    test('should match snapshot', () => {
      const { asFragment } = renderBridgeAction({
        store,
      })
      expect(asFragment()).toMatchSnapshot()
    })

    test('should trigger bridgeConfirmModal on click of bridge', () => {
      renderBridgeAction({ store })
      const bridgeBtn = screen.getByTestId('bridge-btn')
      fireEvent.click(bridgeBtn)
      const actions = store.getActions()
      expect(actions).toEqual([
        {
          destNetworkSelection: undefined,
          fast: undefined,
          lock: undefined,
          payload: 'bridgeConfirmModal',
          proposalId: undefined,
          selectionLayer: undefined,
          token: undefined,
          tokenIndex: undefined,
          type: 'UI/MODAL/OPEN',
        },
      ])
    })

    test('should not invoke bridgeConfirmModal on click of bridge when bridge action disabled', () => {
      alerts = [{ type: 'error', text: 'wrong' }]
      store = mockStore({
        ui: {
          theme: 'dark',
        },
        network: {
          activeNetwork: NETWORK.ETHEREUM,
          activeNetworkType: NETWORK_TYPE.MAINNET,
        },
        bridge: {
          tokens: [],
          amountToBridge: null,
          alerts,
        },
        setup: {
          accountEnabled: true,
        },
      })
      renderBridgeAction({ store })
      const bridgeBtn = screen.getByTestId('bridge-btn')
      fireEvent.click(bridgeBtn)
      const actions = store.getActions()
      expect(actions).toEqual([])
    })
  })

  describe('AccountEnabled is false', () => {
    beforeEach(() => {
      store = mockStore({
        ui: {
          theme: 'dark',
        },
        network: {
          activeNetwork: NETWORK.ETHEREUM,
          activeNetworkType: NETWORK_TYPE.MAINNET,
        },
        bridge: {
          tokens: [{ symbol: 'ETH' }],
          amountToBridge: '1.255',
          alerts: ['alert1', 'alert2'],
        },
        setup: {
          accountEnabled: false,
        },
      })
    })

    test('should match snapshot', () => {
      const { asFragment } = renderBridgeAction({
        store,
      })
      expect(asFragment()).toMatchSnapshot()
    })

    test('should trigger connect action on click of connect wallet', () => {
      renderBridgeAction({ store })
      const connectBtn = screen.getByTestId('connect-btn')
      fireEvent.click(connectBtn)
      const actions = store.getActions()
      expect(actions).toEqual([{ type: 'SETUP/CONNECT', payload: true }])
    })
  })
})
