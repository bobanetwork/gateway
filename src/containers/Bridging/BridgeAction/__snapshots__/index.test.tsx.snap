// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BridgeAction AccountEnabled is false should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR bkXCIc"
  >
    <button
      class="sc-gsFSXq jLgYJo sc-ikkxIA"
      data-testid="connect-btn"
      label="[object Object]"
      style="width: 100%;"
      type="button"
    >
       
      <h3
        class="sc-jlZhew bOWxgm"
      >
        Connect Wallet
      </h3>
    </button>
  </div>
</DocumentFragment>
`;

exports[`BridgeAction AccountEnabled is true should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR bkXCIc"
  >
    <button
      class="sc-gsFSXq jLgYJo sc-ikkxIA"
      data-testid="bridge-btn"
      label="[object Object]"
      style="width: 100%;"
      type="button"
    >
       
      <h3
        class="sc-jlZhew bOWxgm"
      >
        Bridge
      </h3>
    </button>
  </div>
</DocumentFragment>
`;
